fix: Correct test cases and audit commands for rh-1.8, cis-1.11, gke-1.8.0, and eks-1.7.0 benchmarks

amitk1sharma opened the pull request:
This PR fixes several test case inconsistencies and incorrect audit commands in the benchmark YAML files for:

rh-1.8
cis-1.11
gke-1.8.0
eks-1.7.0

Changes made:

Fixed incorrect test case names and descriptions
Updated audit commands
Adjusted scored values to align with CIS standards
Improved remediation steps for clarity
Verified validation paths for OCP, GKE, and EKS consistency

Associated Issue: Fixes #1993

amitk1sharma committed:
Enhance kube-bench configurations

sync cfg updates

sync cfg updates with upstream

Resolve PR comments

comments resolve

PR comments resolved

Removed extra space

Correct spacing and indentation issues

Resolved PR comments

Resolve PR comments

deebhatia commented on cfg/cis-1.11/node.yaml:
```diff
@@ -476,13 +477,14 @@ groups:
+          IPAddressAllow={{ kubelet_secure_addresses }}
+          *Note kubelet_secure_addresses: "localhost link-local {{ kube_pods_subnets | regex_replace(',', ' ') }} {{ kube_node_addresses }} {{ loadbalancer_apiserver.address | default('')"
+          By default IPAddressDeny is not enabled.
         scored: false
```
Keep older text.

deebhatia commented on cfg/eks-1.7.0/controlplane.yaml:
```diff
@@ -9,8 +9,7 @@ groups:
       - id: 2.1.1
-        text: "Enable audit Logs (Manual)"
-        type: manual
+        text: "Enable audit Logs (Automated)"
```
Either add audit section OR keep the type as manual because there is no audit yet. Text can be changed as upstream has it as automated.

deebhatia commented on cfg/eks-1.7.0/node.yaml:
```diff
@@ -375,6 +371,7 @@ groups:
           test_items:
             - flag: --rotate-certificates
               path: '{.rotateCertificates}'
+              set: true
```
Since compare is already present, please revert.

deebhatia commented on cfg/eks-1.7.0/policies.yaml:
```diff
@@ -10,49 +10,21 @@ groups:
+          ClusterRoleBinding to the cluster-admin role :
           kubectl delete clusterrolebinding [name]
-        scored: true
+        scored: false
```
revert this. scored should be true as it is automated.

amitk1sharma replied:
Resolved.

deebhatia commented on cfg/eks-1.7.0/policies.yaml:
```diff
@@ -198,7 +164,7 @@ groups:
       - id: 4.2.2
         text: "Minimize the admission of containers wishing to share the host process ID namespace (Automated)"
-        audit: |
+        taudit: |
```
revert.

amitk1sharma replied:
Resolved.

deebhatia commented on cfg/gke-1.8.0/managedservices.yaml:
```diff
@@ -709,9 +716,8 @@ groups:
-      - id: 5.10.5
+      - id: 5.10.4
         text: "Enable Security Posture (Manual)"
-        audit: "gcloud container clusters --location describe"
```
Keep all the audit commands in this file.

amitk1sharma commented on cfg/eks-1.7.0/controlplane.yaml:
```diff
@@ -34,36 +33,37 @@ groups:
+              name: audit-logging
+              namespace: kube-system
+              spec:
+              containers:
```
Resolved.

amitk1sharma commented on cfg/eks-1.7.0/policies.yaml:
```diff
@@ -10,49 +10,21 @@ groups:
-          test_items:
-            - flag: "SECRETS_ACCESS_FOUND"
-              set: false
+        type: "manual"
```
Resolved.

amitk1sharma commented on cfg/eks-1.7.0/policies.yaml:
```diff
@@ -10,49 +10,21 @@ groups:
         remediation: |
           Where possible, remove get, list and watch access to secret objects in the cluster.
-        scored: true
+        scored: false
```
Resolved.

deebhatia commented on cfg/rh-1.8/master.yaml:
```diff
@@ -212,35 +220,23 @@ groups:
           else
-          # Execute the stat command
-          oc exec -n openshift-sdn "$POD_NAME"  -- find /var/lib/cni/networks/openshift-sdn -type f -exec stat -c "$i %n permissions=%a" {} \; 2>/dev/null
-          oc exec -n openshift-sdn "$POD_NAME"  -- find /var/run/openshift-sdn -type f -exec stat -c "$i %n permissions=%a" {} \; 2>/dev/null
```
the old audit is completely valid. Only fix indentation there..

amitk1sharma commented on cfg/rh-1.8/node.yaml:
```diff
@@ -299,11 +307,10 @@ groups:
           NODE_NAME=$(oc get pod $HOSTNAME -o=jsonpath='{.spec.nodeName}')
-          oc get --raw /api/v1/nodes/$NODE_NAME/proxy/configz \
-            | jq '.kubeletconfig'
+          oc get --raw /api/v1/nodes/$NODE_NAME/proxy/configz | jq '.kubeletconfig.streamingConnectionIdleTimeout' 2> /dev/null
```
Resolved.

amitk1sharma commented on cfg/rh-1.8/master.yaml:
```diff
@@ -560,213 +562,128 @@ groups:
       - id: 1.2.9
         text: "Ensure that the admission control plugin AlwaysPullImages is set (Manual)"
         audit: |
+          #Verify the set of admissi    on-plugins for OCP 4.6 and higher
```
Resolved.

mozillazg reviewed (commented):
@amitk1sharma Thanks for your contribution! I don't have more comments. Please check the comments from @afdesk when you get a chance. Thanks!

andypitcher commented on cfg/cis-1.11/etcd.yaml:
```diff
@@ -27,6 +27,24 @@ groups:
         scored: true
 
       - id: 2.2
+        text: "Ensure that the --cert-file and --key-file arguments are set as appropriate (Automated)"
```
Does it make sense to have twice the same check ? 
<img width="895" height="709" alt="image" src="https://github.com/user-attachments/assets/638f76de-2904-4ea5-8aac-f08f3b7eb3a0" />

After some review, it seems that it was a mistake on CIS Workbench, that was corrected in CIS-1.12.

### CIS-1.11 (and CIS-1.11.1)

<img width="524" height="53" alt="image" src="https://github.com/user-attachments/assets/31bed734-568d-4686-a377-b7fb91f1b983" />


### CIS-1.12

<img width="506" height="40" alt="image" src="https://github.com/user-attachments/assets/3551231e-b260-40d8-bc16-74a1b849a482" />



**With this I don't think it is a good idea to change, as it will be reverted in `CIS-1.12` anyways.**

andypitcher replied:
This would then mean to revert all changes, such as numbering.

amitk1sharma replied:
The duplication in CIS-1.11 (2.1 and 2.2) is intentional to faithfully mirror the published benchmark. Keeping it avoids unnecessary renumbering and churn. Any fixes or updates will naturally appear in the CIS-1.12 profile.

andypitcher replied:
@mozillazg curious to get your thoughts on that front. To me, I'm all for staying as close as possible with CIS content - however replicating know issues that potentially lead to a loss of value and consistency (worst, security) should be avoided to me.

andypitcher commented on cfg/cis-1.11/node.yaml:
```diff
@@ -1,6 +1,6 @@
 ---
 controls:
-version: "cis-1.11"
+version: "cis-1.11.1"
```
```suggestion
version: "cis-1.11"
```

andypitcher commented on cfg/cis-1.11/policies.yaml:
```diff
@@ -1,6 +1,6 @@
 ---
 controls:
-version: "cis-1.11"
+version: "cis-1.11.1"
```
```suggestion
version: "cis-1.11"
```

amitk1sharma commented on cfg/rh-1.8/policies.yaml:
```diff
@@ -13,14 +13,15 @@ groups:
-          oc get clusterrolebindings -o=customcolumns=NAME:.metadata.name,ROLE:.roleRef.name,SUBJECT:.subjects[*].kind |
-          grep cluster-admin
+          oc get clusterrolebindings -o=custom-columns="NAME:.metadata.name,ROLE:.roleRef.name,SUBJECT:.subjects[*].kind" | grep cluster-admin
           #To verity that kbueadmin is removed, no results should be returned
```
Resolved.

amitk1sharma commented on cfg/cis-1.11/etcd.yaml:
```diff
@@ -1,6 +1,6 @@
 ---
 controls:
-version: "cis-1.11"
+version: "cis-1.11.1"
```
Resolved.

amitk1sharma commented on cfg/cis-1.11/master.yaml:
```diff
@@ -1,6 +1,6 @@
 ---
 controls:
-version: "cis-1.11"
+version: "cis-1.11.1"
```
Resolved.

amitk1sharma commented:
Hi @afdesk  @mozillazg, I have resolved the previous comments.
Kindly review the PR again when you get a chance.

amitk1sharma commented:
Hi @afdesk @mozillazg,
I’ve addressed all the previous review comments.
Could you please merge this when you get a chance?

mozillazg reviewed (approved):
LGTM, Thanks for your contribution!

andypitcher reviewed (approved):
Approving the CIS-1.11 part, thanks !

afdesk commented on cfg/rh-1.8/policies.yaml:
```diff
@@ -233,8 +247,13 @@ groups:
           test_items:
-            - flag: "pass"
+            - flag: "MustRunAsNonRoot"
+            - flag: "MustRunAs"
```
it seems that will never match.
could you pls re-check this policy?

afdesk commented on cfg/rh-1.8/etcd.yaml:
```diff
@@ -165,19 +167,19 @@ groups:
               compare:
                 op: regex
-                value: '\/etc\/kubernetes\/static-pod-certs\/configmaps\/(?:etcd-(?:serving|peer-client)-ca\/ca-bundle\.crt|etcd-all-bundles\/server-ca-bundle\.crt)'
+                value: '\/etc\/kubernetes\/static-pod-certs\/configmaps\/etcd-(?:serving|peer-client)-ca\/ca-bundle\.(?:crt|key)'
```
do we really need `key` here? 
it looks like CA bundles are certs, not keys

afdesk commented on cfg/rh-1.8/master.yaml:
ditto
```suggestion
        text: "Ensure that the etcd pod specification file permissions are set to 600 or more restrictive (Manual)"
```

afdesk commented on cfg/gke-1.8.0/managedservices.yaml:
```suggestion
        text: "Ensure GKE clusters are not running using the Compute Engine default service account (Automated)"
```

amitk1sharma replied:
Resolved.

afdesk commented:
> @afdesk Rebase Done. Could you please merge now?

@amitk1sharma thanks for your efforts.
there were made a lot of changes, so I had to re-review this PR again, and left some marks.

Could you pls take a look at them and update?

 I realize that some notes weren't made in your PR, but these typos should be fixed too to improve the project.
Thanks a lot!

amitk1sharma commented:
Hi @afdesk @mozillazg @andypitcher,

Just a gentle follow-up on this PR. 
Please let me know if any further changes are required from my side. 
If everything looks good, could you please merge it when you have a chance?

Thanks!

amitk1sharma commented:
Hi @afdesk @mozillazg @andypitcher,

Just following up on this PR once again. I have updated the branch with the latest changes from main and addressed everything from my side.

Whenever you get a chance, could you please review and help merge it if everything looks good?
Please let me know if any further changes are required.

Thank you!

amitk1sharma commented:
Hi @afdesk @mozillazg, 

Just following up on this PR once again. I have resolved the previous comments.
could you please review and merge it if everything looks good?
Please let me know if any further changes are required.

Thank you!

deebhatia commented:
Hi @afdesk, 
Only your approval is pending. 
Can you please do the same after triggering the checks and once they are green, we can target merge. 
Requesting it at the earliest.
Thanks!

LaibaBareera commented on cfg/rh-1.8/etcd.yaml:
```diff
@@ -16,19 +16,20 @@ groups:
               compare:
                 op: regex
-                value: '\/etc\/kubernetes\/static-pod-certs\/secrets\/etcd-all-(serving|certs)\/etcd-serving-.*\.(?:crt|key)'
+                # some systems have certs in directory '/etc/kubernetes/static-pod-certs/secrets/etcd-all-certs'
```
remove this comment

amitk1sharma replied:
Resolved.

LaibaBareera commented on cfg/rh-1.8/master.yaml:
```diff
@@ -560,213 +562,128 @@ groups:
-          fi
-
-          oc -n openshift-apiserver describe secret serving-cert | grep -E 'tls\.crt|tls\.key|Type:'
+          oc get configmap config -n openshift-kube-apiserver -ojson | jq -r '.data["config.yaml"]' | jq '.apiServerArguments'
```
@amitk1sharma Have you tested this audit command and reviewed its results? As far as I remember, the original audit command was producing false-positive results, which is why I modified it in this way.

amitk1sharma replied:
Yes, I tested the updated command and reviewed its output. It correctly returns the apiServerArguments section, including the kubelet client certificate and key configuration. I have not observed any false-positive behavior during my testing, and the output appears consistent with the cluster configuration.

LaibaBareera reviewed (changes requested):
@amitk1sharma, i have some concerns related modifications in audit command in master.yaml file of rh-1.8

amitk1sharma commented:
Hi @afdesk,

I’ve addressed all the previous review comments.
Could you please re-review and merge it if everything looks good?
Please let me know if any further changes are required.

Thank you!

amitk1sharma commented:
Hi @LaibaBareera,

I’ve addressed all the previous review comments.
Could you please re-review and merge?
Please let me know if any further changes are required.

Thank you!

amitk1sharma commented:
Hi @LaibaBareera ,

I’ve addressed all the previous review comments.
Could you please re-review and merge?

Thank you!
